agent.py

```
@@ -4,8 +4,21 @@
+from langchain.chains import ConversationalRetrievalChai
+from dotenv import load_dotenv
 
-from langchain.chains import ConversationalRetrievalChain
+# setting the environment variable `GOOGLE_APPLICATION_CREDENTIALS` to the file path
+credentials_path = os.environ.get("credentials")
+if credentials_path:
+    os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = credentials_path
+else:
+    # Handle the case where the environment variable is not set
+    # For example, you could raise an exception or use a default value
+    raise ValueError("The 'credentials' environment variable is not set.")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)